src/AirtableClient.php

```diff
@@ -45,58 +68,91 @@ public function findAll(string $table, ?string $view = null): array
      * @param  mixed $table Table name
      * @param  mixed $field Search field name
      * @param  mixed $value Wanted value
+     * @param  string $dataClass The class name which will hold fields data
      * @return array
      */
-    public function findBy(string $table, string $field, string $value): array
+    public function findBy(string $table, string $field, string $value, ?string $dataClass = null): array
     {
-        $filterByFormula = "?filterByFormula=AND({".$field."} = '".$value."')";
-        $url = $this->id .'/'. $table . $filterByFormula;
+        $filterByFormula = "?filterByFormula=AND({" . $field . "} = '" . $value . "')";
+        $url = $this->id . '/' . $table . $filterByFormula;
         $response = $this->request($url);
 
-        return $response->toArray()['records'];
+        $airtableRecords = array_map(
+            fn ($record) => AirtableRecord::fromRecord($record),
+            $response->toArray()['records']
+        );
+
+        if ($dataClass) {
+            foreach ($airtableRecords as $record) {
+                $record->setFields(
+                    $this->normalizer->denormalize(
+                        $record->getFields(),
+                        $dataClass
+                    )
+                );
+            }
+        }
+
+        return $airtableRecords;
     }
-    
+
     /**
      * findOneById
      *
      * @param  mixed $table Table Name
      * @param  mixed $id Id
-     * @return array
+     * @param  string $dataClass The name of the class which will hold fields data 
+     * @return array|object
      */
-    public function findOneById(string $table, string $id): array
+    public function findOneById(string $table, string $id, ?string $dataClass = null)
     {
-        $url = $this->id .'/'. $table . '/' . $id;
+        $url = $this->id . '/' . $table . '/' . $id;
         $response = $this->request($url);
 
-        return $response->toArray();
+        $airtableRecord = AirtableRecord::fromRecord($response->toArray());
+
+        if ($dataClass) {
+            $airtableRecord->setFields(
+                $this->normalizer->denormalize($airtableRecord->getFields(), $dataClass)
+            );
+        }
+
+        return $airtableRecord;
     }
-    
+
     /**
      * findTheLatest
      *
      * Field allowing filtering
      *
      * @param  mixed $table Table name
      * @param  mixed $field
-     * @return array
+     * @param  string $dataClass The name of the class which will hold fields data
+     * @return AirtableRecord
      */
-    public function findTheLatest(string $table, $field): array
+    public function findTheLatest(string $table, $field, ?string $dataClass = null): AirtableRecord
```

Comment:
can be null if there is no result `?AirtableRecord`

Reply:
Fixed !